### PR TITLE
Remove 'OK' from HTTP response when no 'body' is present.

### LINKED ISSUE #2132: Web action with http extension defining "OK" when body is missing from response
Create a web action which doesn't return a "body" top level property and curl it as a web action with .http as the extension. It responds with "OK" which comes from the lower spray layer that binds a message to the status code 200.

### PR DESCRIPTION
**Pull request by rabbah**
Fixes #2132.

### CODE REVIEW

**Comment by csantanapr** [comment]
PG?

**Comment by rabbah** [comment]
Not needed but will launch if you like.

**Comment by dubeejw** [comment]
@rabbah, there is some discussion around not returning an HTTP code of 200 when the response is empty.

**Comment by rabbah** [comment]
I reject that as non-nonsensical but willing to debate it if someone wants to take the opposing view.
 
Per https://github.com/openwhisk/openwhisk/issues/2132#issuecomment-293562946, the implementation follows the defaults and I updated the missing default in the docs for body.

**Comment by rabbah** [comment]
pg1/1411 approved.

File: docs/webactions.md [hunk 1]
```diff
@@ -88,7 +88,7 @@ An OpenWhisk action that is not a web action requires both authentication and mu
 
 1. `headers`: a JSON object where the keys are header-names and the values are string values for those headers (default is no headers).
 2. `statusCode`: a valid HTTP status code (default is 200 OK).
-3. `body`: a string which is either plain text or a base64 encoded string (for binary data).
+3. `body`: a string which is either plain text or a base64 encoded string for binary data (default is empty response).
```
Comment (akrabat) [inline docs/webactions.md]: Is this true?

if I `return ["statusCode": 405]`, I'll get a `text/plain` response with the body set to `A request was made of a resource using a request method not supported by that resource;`.

i.e. default is an empty response only in the case of a `200` status code.

Comment (rabbah) [inline docs/webactions.md]: With this patch? I will add an explicit test for this but a quick sniff test:

```
> curl -vk https://guest.openwhisk/default/t
*   Trying 192.168.99.100...
* Connected to guest.openwhisk (192.168.99.100) port 443 (#0)
* TLS 1.2 connection using TLS_ECDHE_RSA_WITH_RC4_128_SHA
* Server certificate: *.openwhisk
> GET /default/t HTTP/1.1
> Host: guest.openwhisk
> User-Agent: curl/7.43.0
> Accept: */*
> 
< HTTP/1.1 405 Method Not Allowed
< Server: nginx/1.11.13
< Date: Mon, 17 Apr 2017 23:54:07 GMT
< Content-Length: 0
< Connection: keep-alive
< 
* Connection #0 to host guest.openwhisk left intact
```

for this action:
```
function main() { return { statusCode: 405 } }
```

Comment (akrabat) [inline docs/webactions.md]: You are correct - I was on the wrong branch when I tested…

**Pull request merged**